app/assets/src/components/common/CoverageVizSidebar/CoverageVizSidebar.jsx

```diff
@@ -0,0 +1,367 @@
+};
+
+const METRIC_TOOLTIPS = {
+  referenceAccession: "Genbank Accession ID for the reference accession.",
+  referenceLength: "Length in bp of the reference accession sequence.",
+  alignedContigs:
+    "Number of contigs for which this accession was the best match.",
+  avgAlignedLength:
+    "Length of aligned regions, averaged over all reads and contigs ",
+  avgCoverageDepth:
+    "The total length of all aligned contigs and reads, divided by the accession length.",
+  coverageBreadth:
+    "The percentage of the accession that is covered by at least one read or contig.",
+  alignedReads: "Number of reads for which this accession was the best match.",
+  avgMismatchedPercent:
+    "Percentage of aligned regions that are mismatches, averaged over all reads and contigs."
+};
+
+const METRIC_COLUMNS = [
+  [
+    {
+      key: "referenceAccession",
+      name: "Reference Accession"
+    },
+    {
+      key: "referenceLength",
+      name: "Reference Length"
+    }
+  ],
+  [
+    {
+      key: "alignedContigs",
+      name: "Aligned Contigs"
+    },
+    {
+      key: "avgAlignedLength",
+      name: "Avg. Aligned Length"
+    }
+  ],
+  [
+    {
+      key: "avgCoverageDepth",
+      name: "Avg. Coverage Depth"
+    },
+    {
+      key: "coverageBreadth",
+      name: "Coverage Breadth"
+    }
+  ],
+  [
+    {
+      key: "alignedReads",
+      name: "Aligned Reads"
+    },
+    {
+      key: "avgMismatchedPercent",
+      name: "Avg. Mismatched %"
+    }
+  ]
+];
+
+const alignRange = range => [min(range), max(range)];
+
+export default class CoverageVizSidebar extends React.Component {
+  _accessionDataCache = {};
+
+  state = {
+    currentAccession: null
+  };
+
+  componentDidUpdate(prevProps, prevState) {
+    if (!prevProps.visible && this.props.visible) {
+      this.setCurrentAccession(this.props.params.accessionSummaries[2].id);
+    }
+
+    if (!prevState.currentAccessionData && this.state.currentAccessionData) {
+      this.renderHistogram(this.state.currentAccessionData);
+      this.renderGenomeViz(this.state.currentAccessionData);
+    }
+  }
+
+  getDataForAccession = async accessionId => {
+    if (this._accessionDataCache[accessionId]) {
+      return this._accessionDataCache[accessionId];
+    } else {
+      // Replace with network fetch.
+      const data = ACCESSION_COVERAGE_DATA[accessionId];
+
+      this._accessionDataCache[accessionId] = data;
+      return data;
+    }
+  };
+
+  loadAccession = async accession => {
+    const data = await this.getDataForAccession(accession.id);
+
+    this.setState({
+      currentAccessionData: data
+    });
+  };
+
+  setCurrentAccession = accessionId => {
+    const accession = find(
+      ["id", accessionId],
+      this.props.params.accessionSummaries
+    );
+
+    this.setState({
+      currentAccession: accession,
+      currentAccessionData: null
+    });
+    this.loadAccession(accession);
+  };
+
+  processCoverageData = (coverageData, coverageBinSize) =>
```

Comment:
I'd prefer something more specific than `processCoverageData` such as `getBinsPositionAndHeight`

Reply:
I'll do `generateCoverageVizData`, since we are generating the data for the coverage viz. "bins position and height" is too specific and more confusing, I think.